### ISSUE TITLE
关于Integral函数的意义

### ISSUE THREAD

**Issue by zxf864823150** [RangiLyu/nanodet#393]
我发现在前项中
`
class Integral(nn.Module):
    def __init__(self, reg_max=16):
        super(Integral, self).__init__()
        self.reg_max = reg_max
        self.register_buffer(
            "project", torch.linspace(0, self.reg_max, self.reg_max + 1)
        )

    def forward(self, x):
        print("x shape is {}".format(x.shape))
        x = F.softmax(x.reshape(-1, self.reg_max + 1), dim=1)
        print("x softmax shape is {}".format(x.shape))
        print("self.project {}".format(self.project))
        x = F.linear(x, self.project.type_as(x)).reshape(-1, 4)
        print("x F.liner is {}".format(x))
        return x
`
其中liner的参数是一个tensor([0,1,2,3,4,5,6,7])定值，输入的feature是52*52*32，我想知道x = F.linear(x, self.project.type_as(x)).reshape(-1, 4)这一步，将softmax后的结果内积这样一个向量的实际意思是什么？

**Comment by RangiLyu** [comment]
Please refer to DFL in https://arxiv.org/pdf/2006.04388.pdf
